### ISSUE TITLE
How should I build a game for wasm

### ISSUE THREAD

**Issue by gmorenz** [not-fl3/macroquad#5]
I feel like I must be doing something wrong.

Cargo refuses to build a binary with `cargo build --target wasm32-unknown-unknown`.

I can build as a library instead, then I get an error about main not being exported when I try to load the library in html.

I can fix that error by applying the following patch, which works fine, but I don't understand how anyone is using this library as-is. What might I be doing wrong/differently?

```
diff --git a/macroquad_macro/src/lib.rs b/macroquad_macro/src/lib.rs
index f8496a5..4b28673 100644
--- a/macroquad_macro/src/lib.rs
+++ b/macroquad_macro/src/lib.rs
@@ -36,6 +36,7 @@ pub fn main(_attr: TokenStream, item: TokenStream) -> TokenStream {
 
     let mut prelude: TokenStream = format!(
         "
+    #[cfg_attr(target_arch = \"wasm32\", no_mangle)]
     fn main() {{
         macroquad::Window::new(\"MACROQUAD\", amain());
     }}
```

**Comment by not-fl3** [comment]
Hi! 
Just tried with an empty project - works for me. 
Could you try to run this project and post errors if any?

Clone https://github.com/not-fl3/macroquad-sample-project and run `cargo build --target wasm32-unknown-unknown`

**Comment by gmorenz** [comment]
Hmm, yep, I agree that empty project works... and is set up in the obvious way. Will try and figure out what is different about what I did.

**Comment by gmorenz** [comment]
Alright, figured this out. It was entirely my fault, sorry for wasting your time.

Before adding macroquad I was building a wasm library and a non-wasm binary. As a result I had `[lib] crate-type = ["cdylib", "rlib"]` in my Cargo.toml. When I switched to macroquad I was apparently continuing to use the library version of the wasm when building both (which doesn't export main), and when I tried removing `lib.rs` it refused to build at all because I had already told it to expect a library.
